File: 2021/0121_maxProfit/0824/maxProfit.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lowestPrice, maxProfit = float("inf"), 0
        for p in prices:
            lowestPrice = min(p, lowestPrice) 
            maxProfit = max(maxProfit, p - lowestPrice)
        return maxProfit

File: 2021/0121_maxProfit/0824/maxProfit.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let (mut lowest_price, mut max_profit) = (std::i32::MAX, 0);
        for &p in prices.iter() {
            lowest_price = std::cmp::min(lowest_price, p);
            max_profit = std::cmp::max(max_profit, p - lowest_price);
        }
        max_profit
    }
}
